### PR TITLE
Add org.openrgb.OpenRGB

### PR DESCRIPTION
**Pull request by tim77**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://gitlab.com/CalcProgrammer1/OpenRGB/-/issues/1179
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - https://gitlab.com/CalcProgrammer1/OpenRGB/-/merge_requests/933
  - https://gitlab.com/CalcProgrammer1/OpenRGB/-/commit/ea00ca89770b2143ccae05183da3b1b5f0e36710
  - https://gitlab.com/CalcProgrammer1/OpenRGB/-/merge_requests/934
  - https://gitlab.com/CalcProgrammer1/OpenRGB/-/merge_requests/935

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

---

Based on stalled https://github.com/flathub/flathub/pull/2508.

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.openrgb.OpenRGB

**Comment by tim77** [comment]
There is no appdata. OK. I'll try to write and submit upstream.

**Comment by hfiguiere** [comment]
> There is no appdata. OK. I'll try to write and submit upstream.

I did mention it :-) https://github.com/flathub/flathub/pull/2508#issuecomment-920450156

**Comment by tim77** [comment]
bot, build

**Comment by tim77** [comment]
bot, build org.openrgb.OpenRGB

**Comment by tim77** [comment]
bot, build org.openrgb.OpenRGB

**Comment by tim77** [comment]
Icon broken since it installed not in proper location. Now it's in `share/pixmaps/`. Options:
1. Fix icon in downstream for now.
2. Try to fix in upstream. This will requires to fix all hardcoded paths in upstream CI for various distros and others builds/packages. Tedious, but better (maybe).

**Comment by nedrichards** [comment]
Why not both? Pragmatism is good and helps us get things faster but unless fixes get upstream the effort to keep supporting things will eventually get too much.

File: org.openrgb.OpenRGB.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+id: org.openrgb.OpenRGB
+sdk: org.kde.Sdk
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+command: openrgb
+finish-args:
+  - --device=all
+  - --share=ipc
+  - --socket=wayland
+  - --socket=x11
```
[the remark this answers is not in the document]
Comment (tim77) [reply org.openrgb.OpenRGB.yaml:10]: Right, didn't have time to test on wayland yet.

**Comment by tim77** [comment]
bot, build org.openrgb.OpenRGB

**Comment by nedrichards** [comment]
I don't have supported hardware so @tim77 can you confirm that this package actually works ;-) other than that this lgtm

**Comment by tim77** [comment]
@nedrichards i believe there will be some limitations with supported hardware because some of them requires udev rules. I'll try to discuss with upstream what we can do there: provide README for flatpak version or document this somewhere in official wiki or even some tip in app itself. But in same time upstream provide AppImage under the Releases section on GitLab but i suppose it should have the same limitations since udev rules should be installed on the host...

Any help with testing flatpak build very welcome and needed.

**Comment by nedrichards** [comment]
Right, if it's user visible restrictions then those should go into the appdata description - that's where users will actually see it. A README in git is functionally invisible to them. As you say, tips in app are even better.

**Comment by stuarthayhurst** [comment]
`flatpak install --user https://dl.flathub.org/build-repo/68621/org.openrgb.OpenRGB.flatpakref` gave me:
```
error: Can't load uri https://dl.flathub.org/build-repo/68621/org.openrgb.OpenRGB.flatpakref: Server returned status 404: Not Found
```

**Comment by hfiguiere** [comment]
bot, build org.openrgb.OpenRGB

**Comment by hfiguiere** [comment]
build expired. triggering a new one.

**Comment by stuarthayhurst** [comment]
image
No devices were detected, it should pick up my Logitech G703 mouse and Gigabyte B550 motherboard... strangely, when the Debian package is installed, the Debian package works and the flatpak version magically starts working. Is this down to the `udev` rules that were mentioned earlier?

**Comment by stuarthayhurst** [comment]
Yep, that was it. Copying `60-openrgb.rules` from the project to `/etc/udev/rules.d` and reloading with `sudo udevadm control --reload-rules && sudo udevadm trigger` now allows the flatpak to find the devices. However, when I click rescan devices the app crashes, and the regular Debian build doesn't.

**Comment by tim77** [comment]
@stuarthayhurst great, thanks a lot for feedback. That exactly what we) thought. I am also wanted to test RPM version and see if it works and ask some folks to test it. But what is weird for me that i believe current official AppImage will have the same problems with udev rules... Upstream discussion for udev things [here](https://gitlab.com/CalcProgrammer1/OpenRGB/-/merge_requests/933#note_793816683).

**Comment by stuarthayhurst** [comment]
Yep, the AppImage has the exact same issues, I tired that originally and then tried the .deb, which worked because it shipped the rules.

**Comment by FakeShemp** [comment]
0.7 has been released.

**Comment by tim77** [comment]
bot, build org.openrgb.OpenRGB

**Comment by FakeShemp** [comment]
Works for me :+1:

**Pull request closed without merging**